cyclient/scan_client.py

```diff
@@ -1,86 +1,87 @@
 import json
+from typing import List
+
 import requests.exceptions
 from requests import Response
-from typing import List
-from . import models
-from .cycode_token_based_client import CycodeTokenBasedClient
-from cli.zip_file import InMemoryZip
+
 from cli.exceptions.custom_exceptions import CycodeError, HttpUnauthorizedError
-from cyclient import logger
+from cli.zip_file import InMemoryZip
+from . import models
+from .cycode_client_base import CycodeClientBase
+from .scan_config import ScanConfigBase
 
 
 class ScanClient:
-    SCAN_CONTROLLER_PATH = 'api/v1/scan'
-    SCAN_SERVICE_CONTROLLER_PATH = 'scans/api/v1/scan'
-    DETECTIONS_SERVICE_CONTROLLER_PATH = 'detections/api/v1/detections'
-
-    def __init__(self, client_id: str = None, client_secret: str = None):
-        self.cycode_client = CycodeTokenBasedClient(client_id, client_secret)
+    def __init__(self, scan_cycode_client: CycodeClientBase, detection_cycode_client: CycodeClientBase,
```

Comment:
need to think on better way than creating two clients

Reply:
I did it due to localhost issue which sending request to scan service with port 5004 and detection with port 5016